### PR TITLE
Fixes #10230 - Host clone uses create_hosts permission

### PR DESCRIPTION
**Pull request by dLobatog**
Non-admin users are not able to clone hosts. This is a regression, in
the past the create_hosts permission has been used for this, but it no
longer works. Passing the permission directly fixes the regression.

### CODE REVIEW

**Comment by tbrisker** [comment]
This works for me. 
However, this isn't the only place we have a clone action. 
Other places (such as hostgroup or taxonomy) work without passing the permission directly - what is the difference here that breaks it?

**Comment by dLobatog** [comment]
@tbrisker If I remember correctly the issue is that we the "Host::Managed" not the "Host" class, hence the _hosts permissions don't get picked up.

**Comment by domcleal** [comment]
It's a difference based on what they're authorising on.  The host list is authorising a host object (the one being cloned), as it sets `:auth_object => host`, but the host group list is authorising based on the controller action only.

Cloning is a mix of operations - you need view_\* on the object you're cloning, and create_\* to access the form (defined in the controller's action_permission method, and app/services/foreman/access_permissions.rb respectively.)

Since if you can see the host in the list, you implicitly have the view_hosts permission on it, I think the authorisation check should be copied from host groups, which simply shows the link if you have access to the HostgroupsController#clone route.

In short:

```
display_link_if_authorized(_("Clone"), hash_for_clone_host_path(:id => host)),
```

should be enough.  I think it's pretty much the same, but more consistent?

**Comment by dLobatog** [comment]
@domcleal @tbrisker Updated, now with 30% less code :smile:

**Comment by domcleal** [comment]
Merged as 3f13099, thanks @eLobato.
